src/ceph-daemon/ceph-daemon

```diff
@@ -1616,6 +1616,105 @@ def command_rm_cluster():
 
 ##################################
 
+def check_time_sync():
+    units = ['chronyd.service', 'systemd.timesyncd', 'ntpd.service']
+    for u in units:
+        (enabled, state) = check_unit(u)
+        if enabled and state == 'running':
+            logger.info('Time sync unit %s is enabled and running' % u)
+            return True
+    return False
+
+def check_firewalld(fix=False):
+    cmd = find_program('firewall-cmd')
+    if not cmd:
+        logger.info('firewalld does not appear to be present')
+        return True  # no firewalld -> all good
+    (enabled, state) = check_unit('firewalld.service')
+    if not enabled:
+        logger.info('firewalld.service is not enabled')
+        return True
+    for svc in ['ceph', 'ceph-mon']:
+        out, err, ret = call([cmd, '--permanent', '--query-service', svc])
+        if ret:
+            if not fix:
+                logger.error('firewalld service %s is not enabled in current zone' % svc)
+                return False
+            logger.info('Enabling firewalld service %s in current zone...' % svc)
+            out, err, ret = call([cmd, '--permanent', '--add-service', svc])
+            if err:
+                logger.error('unabled to add service %s to current zone' % svc)
+                return False
+        else:
+            logger.info('firewalld service %s is enabled in current zone' % svc)
+    return True
+
+def command_check_host():
+    # caller already checked for docker/podman
+    logger.info('podman|docker (%s) is present' % container_path)
+
+    if not find_program('systemctl'):
+        raise RuntimeError('unable to location systemctl')
+    logger.info('systemctl is present')
+
+    if not find_program('lvcreate'):
+        raise RuntimeError('LVM does not appear to be installed')
+    logger.info('LVM2 is present')
+
```

Comment:
why not put the programs to look for in a list as a global so check and prepare can both act on the same stuff?

Reply:
it's never as simple as a package list, since (1) we arent detecting distro and (2) package names vary and (3) for the 'prepare' mode we are opinionated (e.g., always chrony, but check is happy with ntpd and systemd-timesyncd too).